clouddriver-aws/src/test/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/loadbalancer/UpsertAmazonLoadBalancerV2AtomicOperationSpec.groovy

```diff
@@ -209,6 +343,42 @@ class UpsertAmazonLoadBalancerV2AtomicOperationSpec extends Specification {
     0 * _
   }
 
+  void "should modify only target group attributes that are passed of existing load balancer"() {
+    @Subject updateOperation = new UpsertAmazonLoadBalancerV2AtomicOperation(updateDescription)
+
+    setup:
+    def existingLoadBalancers = [ loadBalancerOld ]
+    def existingTargetGroups = [ targetGroup ]
+    def existingListeners = []
+    updateOperation.amazonClientProvider = mockAmazonClientProvider
+    updateOperation.regionScopedProviderFactory = regionScopedProviderFactory
+    updateOperation.deployDefaults = new AwsConfiguration.DeployDefaults(addAppGroupToServerGroup: true, createLoadBalancerIngressPermissions: true)
+    updateOperation.ingressLoadBalancerBuilder = ingressLoadBalancerBuilder
+
+    when:
+    updateOperation.operate([])
+
+    then:
+    1 * loadBalancing.describeLoadBalancers(new DescribeLoadBalancersRequest(names: ["foo-main-frontend"])) >>
+      new DescribeLoadBalancersResult(loadBalancers: existingLoadBalancers)
+    1 * loadBalancing.setSecurityGroups(new SetSecurityGroupsRequest(
+      loadBalancerArn: loadBalancerArn,
+      securityGroups: ["sg-1234"]
+    ))
+    1 * loadBalancing.describeTargetGroups(new DescribeTargetGroupsRequest(loadBalancerArn: loadBalancerArn)) >> new DescribeTargetGroupsResult(targetGroups: existingTargetGroups)
+    1 * loadBalancing.modifyTargetGroup(_ as ModifyTargetGroupRequest)
+    1 * loadBalancing.modifyTargetGroupAttributes(_) >> { ModifyTargetGroupAttributesRequest request ->
+      assert request.attributes.find { it.key == 'deregistration_delay.timeout_seconds'}.value == "300"
```

Comment:
assert only attributes sent in the description set are present during the update call

Reply:
similar nit comment.